### PR TITLE
Document evolution scene

### PR DESCRIPTION
**Pull request by GriffinRichards**
Label unnamed functions/data in `evolution_scene` and `evolution_graphics`, and adjacent documentation / clean up

### CODE REVIEW

File: src/pokemon.c [hunk 1]
```diff
@@ -2993,7 +2993,7 @@ void GiveBoxMonInitialMoveset(struct BoxPokemon *boxMon)
 
 u16 MonTryLearningNewMove(struct Pokemon *mon, bool8 firstMove)
 {
-    u32 retVal = 0;
+    u32 retVal = MOVE_NONE;
```
Comment (jiangzhengwenjz) [inline src/pokemon.c:2996]: This variable is unused

File: src/evolution_scene.c [hunk 2]
```diff
@@ -878,350 +937,403 @@ static void Task_EvolutionScene(u8 taskID)
...
         {
-            InitMovingBackgroundTask(TRUE);
-            var = gSprites[sEvoStructPtr->preEvoSpriteID].oam.paletteNum + 16;
-            sEvoGraphicsTaskID = LaunchTask_PreEvoSparklesSet1(var);
-            gTasks[taskID].tState++;
+            StartBgAnimation(TRUE);
+            var = gSprites[sEvoStructPtr->preEvoSpriteId].oam.paletteNum + 16;
+            sEvoGraphicsTaskId = EvolutionSparkles_SpiralUpward(var);
+            gTasks[taskId].tState++;
             SetGpuReg(REG_OFFSET_BG3CNT, 0x603);
```
Comment (jiangzhengwenjz) [inline src/evolution_scene.c:1127]: should be using bg cnt constants

File: src/pokemon.c (hunk 1 above)
Comment (GriffinRichards) [reply src/pokemon.c:2996]: It's used below

File: src/evolution_scene.c (hunk 2 above)
Comment (GriffinRichards) [reply src/evolution_scene.c:1127]: There's lots of missing constant usage. This has been merged, if you'd like to add the constant usage please open a PR

File: include/evolution_graphics.h [hunk 3]
```diff
@@ -3,13 +3,11 @@
...
 
-u8 LaunchTask_PreEvoSparklesSet1(u16 arg0);
-u8 LaunchTask_PreEvoSparklesSet2(void);
-u8 LaunchTask_PostEvoSparklesSet1(void);
-u8 LaunchTask_PostEvoSparklesSet2AndFlash(u16 species);
-u8 LaunchTask_PostEvoSparklesSet2AndFlash_Trade(u16 species);
-u8 sub_817C3A0(u8 preEvoSpriteID, u8 postEvoSpriteID);
-
-#define EvoGraphicsTaskEvoStop  data[8]
+u8 EvolutionSparkles_SpiralUpward(u16 arg0);
```
Comment (jiangzhengwenjz) [inline include/evolution_graphics.h:6]: arg0->palNum

File: src/pokemon.c (hunk 1 above)
Comment (jiangzhengwenjz) [reply src/pokemon.c:2996]: You are right and I was blind

**Pull request merged**